app/controllers/application_controller.rb

```diff
@@ -167,6 +167,20 @@ def colorize(object)
   end
   helper_method :colorize
 
+  # the identity authenticated for the current session
+  # @example get the okta user id
+  #   current_user_identity&.uid
+  # @return [OauthIdentity, nil]
+  def current_user_identity
+    return nil unless current_user
```

Comment:
maybe?
```suggestion
    return @current_user_identity if @current_user_identity.present?
    return nil unless current_user
```

Reply:
Otherwise we're memoizing it, but doing all the work to re-calculate on each call.  That said, this won't preserve it across page loads, so maybe it doesn't matter.  Should we store it in the session maybe?

Reply:
@eanders Thanks for the feedback! Happy to make that change as a micro optimization. 

Something to note: I suspect this will only ever be called once or twice per request; I just have a habit of memoizing db queries inside of view helpers. Perhaps we just shouldn't use memoization here at all.

--------------

Re moving this into a session: You mean storing the active_provider in session instead of a cookie? That would increase the scope of this issue. We use a cookie since the value is needed after the user logs out so we can redirect them correctly. The session is reset on logout so we'd need to copy this var, it's a bit of extra plumbing that might require QA. Happy to do that if you think it's worth it to eliminate the cookie